src/main/java/ca/wolfram/beta/symath/operations/AddNode.java

```diff
@@ -35,19 +35,49 @@ public NodeType getType() {
         return NodeType.ADD;
     }
 
+    /**
+     * Extracts all constant integers and appends it as one {@link ca.wolfram.beta.symath.base.ConstantNode} if the sum != 0
+     * Flattens children if they are also of type ADD
+     *
+     * @return true is isConstant, false otherwise
+     */
     @Override
     public boolean simplify() {
-        int constant = 0;
-        final Iterator<MathNode> each = getChildren().iterator();
+        int constant = simplify(0, 0);
+        if (constant != 0)
+            getChildren().add(BaseNode.create(constant));
+        return super.simplify();
+    }
+
+    /**
+     * Helper method to help with simplification
+     * Starts the listIterator at index, with history of constant, and continues from there
+     *
+     * @param index    children index to start iteration
+     * @param constant integer summation of child nodes before index
+     * @return new integer summation
+     */
+    private int simplify(int index, int constant) {
+        List<MathNode> childrenToRaise = new ArrayList<>();
+        final ListIterator<MathNode> each = getChildren().listIterator(index);
         while (each.hasNext()) {
             MathNode next = each.next();
-            if (!next.isConstant() || !MathUtils.isConstantInt(next)) continue;
-            each.remove();
-            constant += next.eval(null);
+            if (next.getType() == NodeType.ADD) {
+                childrenToRaise.addAll(next.getChildren());
+                each.remove();
+                continue;
+            }
+            if (MathUtils.isConstantInt(next)) {
+                each.remove();
+                constant += next.eval(null);
+            }
         }
-        MathNode newConstant = BaseNode.create(constant);
-        getChildren().add(newConstant);
-        return super.simplify();
+        if (!childrenToRaise.isEmpty()) {
+            int last = getChildren().size();
+            getChildren().addAll(childrenToRaise);
+            return simplify(last, constant); //recurse and return total constant sum
```

Comment:
I just removed those comments altogether. Generally everything should be easy to understand, and we only need a comment header. We can also use the C++ conventions since the ones [here](http://www.oracle.com/technetwork/java/codeconventions-141999.html) don't seem that good.